mlflow/models/resources/agent_evaluation_template.html

```diff
@@ -0,0 +1,261 @@
+            d="M8 8.917 10.947 6 12 7.042 8 11 4 7.042 5.053 6z"
+            clip-rule="evenodd"
+          ></path>
+        </svg>
+      </span>
+    </button>
+  </div>
+  <div id="code" style="display: none" onclick="showCode()">
+    <h1>
+      Agent evaluation
+      <a
+        class="link"
+        href="https://docs.databricks.com/en/generative-ai/agent-evaluation/synthesize-evaluation-set.html?utm_source=mlflow.log_model&utm_medium=notebook"
+        target="_blank"
+      >
+        <span class="link-content">
+          Learn more
+          <span role="img" aria-hidden="true" class="anticon css-6xix1i"
+            ><svg
+              xmlns="http://www.w3.org/2000/svg"
+              width="1em"
+              height="1em"
+              fill="none"
+              viewBox="0 0 16 16"
+              aria-hidden="true"
+              focusable="false"
+              class=""
+            >
+              <path
+                fill="currentColor"
+                d="M10 1h5v5h-1.5V3.56L8.53 8.53 7.47 7.47l4.97-4.97H10z"
+              ></path>
+              <path
+                fill="currentColor"
+                d="M1 2.75A.75.75 0 0 1 1.75 2H8v1.5H2.5v10h10V8H14v6.25a.75.75 0 0 1-.75.75H1.75a.75.75 0 0 1-.75-.75z"
+              ></path></svg></span></span
+      ></a>
+    </h1>
+    <p class="info">
+      Copy the following code snippet in a notebook cell (right click → copy)
+    </p>
+    <div class="tabs">
+      <div class="tab active" onclick="tabClicked(0)">Using synthetic data</div>
+      <div class="tab" onclick="tabClicked(1)">Using your own dataset</div>
+    </div>
+    <div style="height: 400px">
+      <pre
+        class="active"
+      ><code class="language-python">%pip install -U databricks-agents
+
+from databricks.agents.evals import generate_evals_df
+agent_description = "A chatbot that answers questions about Databricks."
+question_guidelines = """
+# User personas
+- A developer new to the Databricks platform
+# Example questions
+- What API lets me parallelize operations over rows of a delta table?
+"""
+
+# TODO: Spark/Pandas DataFrame with "content" and "doc_uri" columns.
+docs=spark.table("catalog.schema.my_table_of_docs")
+
+evals = generate_evals_df(
+  docs=docs, num_evals=25, agent_description=agent_description, question_guidelines=question_guidelines)
+
+eval_result = mlflow.evaluate(
+    data=evals,
+    model="{{ modelUri }}",
+    model_type="databricks-agent")</code></pre>
+
+      <pre><code class="language-python">%pip install -U databricks-agents
+
+import pandas as pd
+
+evals =  [{
+  "request": {
+    "messages": [{
+      "role": "user",
+      "content": "How do I convert a Spark DataFrame to Pandas?"
+    }],
+  },
+  # Optional, needed for judging correctness.
+  "expected_facts": ["To convert a Spark DataFrame to Pandas, you can use the toPandas() method."]
+}]
+
+eval_result = mlflow.evaluate(
+    data=pd.DataFrame.from_records(evals),
+    model="{{ modelUri }}",
```

Comment:
Can we save this code in a `.py` file? That would allow us to lint the file and prevent syntax errors.

Reply:
Done. Had to pull out "%pip install" to avoid ruff syntax errors and added `# ruff: noqa: F821` on top of file to avoid ruff complaining about `dbutils` and `spark` being undefined. I remove that comment before displaying the html